src/Microsoft.Azure.EventHubs/Amqp/AmqpPartitionReceiver.cs

```diff
@@ -130,15 +130,16 @@ protected override void OnSetReceiveHandler(IPartitionReceiveHandler newReceiveH
         {
             lock (this.receivePumpLock)
             {
-                if (newReceiveHandler != null && this.receiveHandler != null)
+                if (newReceiveHandler != null)
                 {
-                    // Notify existing handler first (but don't wait).
-                    this.receiveHandler.ProcessErrorAsync(new OperationCanceledException("New handler has registered for this receiver."));
-                }
+                    if (this.receiveHandler != null)
+                    {
+                        // Notify existing handler first (but don't wait).
+                        this.receiveHandler.ProcessErrorAsync(new OperationCanceledException("New handler has registered for this receiver."));
```

Comment:
>ProcessErrorAsync [](start = 44, length = 17)

at the very least we should catch and ignore any exception from user handler so that we don't end up in situation where we have not set the new handler.